.stylelintrc.json

```diff
@@ -1,6 +1,482 @@
+        ]
+      },
+      {
+        "groupName": "animation",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "animation",
+          "animation-name",
+          "animation-duration",
+          "animation-timing-function",
+          "animation-delay",
+          "animation-iteration-count",
+          "animation-direction",
+          "animation-fill-mode",
+          "animation-play-state"
+        ]
+      },
+      {
+        "groupName": "vertical-alignment",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "vertical-align"
+        ]
+      },
+      {
+        "groupName": "text-alignment & decoration",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "direction",
+          "tab-size",
+          "text-align",
+          "text-align-last",
+          "text-justify",
+          "text-indent",
+          "text-transform",
+          "text-decoration",
+          "text-decoration-color",
+          "text-decoration-line",
+          "text-decoration-style",
+          "text-rendering",
+          "text-shadow",
+          "text-overflow"
+        ]
+      },
+      {
+        "groupName": "text-spacing",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "line-height",
+          "word-spacing",
+          "letter-spacing",
+          "white-space",
+          "word-break",
+          "word-wrap",
+          "color"
+        ]
+      },
+      {
+        "groupName": "font",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "font",
+          "font-family",
+          "font-size",
+          "font-size-adjust",
+          "font-stretch",
+          "font-weight",
+          "font-smoothing",
+          "osx-font-smoothing",
+          "font-variant",
+          "font-style"
+        ]
+      },
+      {
+        "groupName": "content",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "content",
+          "quotes"
+        ]
+      },
+      {
+        "groupName": "counters",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "counter-reset",
+          "counter-increment"
+        ]
+      },
+      {
+        "groupName": "breaks",
+        "emptyLineBefore": "always",
+        "noEmptyLineBetween": true,
+        "properties": [
+          "page-break-before",
+          "page-break-after",
+          "page-break-inside"
+        ]
+      }
+    ],
+    "max-nesting-depth": [
+      3,
+      {
+        "ignoreAtRules": [
+          "each",
+          "media",
+          "supports",
+          "include"
+        ]
+      }
+    ],
+    "scale-unlimited/declaration-strict-value": [
+      "/color$/",
```

Comment:
For now, we only want all `color` properties to have variable. If we want, we can pass others as well.